<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
    <script src="../lib/jquery.js"></script>
</head>
<body>

</body>
<script>

    const pro1 = new Promise((resolve, reject) => {
        resolve('success');
    }).then((res) => { // 正常情况下，只有上面执行了resolve 才会来到这里, 这里的res 就是上面resolve 的入参
        console.log(res);
        return new Promise((resolve, reject) => {
            resolve('success in another promise'); // 如果返回一个promise对象，这里必须是调用resolve才能到下面的then,
        });
    }).then((res) => { // 但是如果上面返回的是一个promise对象，这个promise 对象必须执行resolve 才能到这里的res, res的值是resolve的入参
        console.log(res);
        return 'success again and again';
    }).then((res) => {
        return res;
    });


</script>
</html>